snapcraft_legacy/plugins/v2/_ros.py

```diff
@@ -76,6 +79,41 @@ def _get_build_commands(self) -> List[str]:
         specific functionality.
         """
 
+    def _get_list_packages_commands(self) -> List[str]:
+
+        cmd = list()
+
+        # Clean up previously established list of packages in build snaps
+        cmd.append('rm -f "${SNAPCRAFT_PART_INSTALL}/.installed_packages.txt"')
+        cmd.append('rm -f "${SNAPCRAFT_PART_INSTALL}/.build_snaps.txt"')
+
+        if self.options.build_snaps:
```

Comment:
Here too we should use the run-time snap instead of the build snaps.
It also means that we will have to download the runtime snaps at build time